serializers/src/glTF/2.0/babylon.glTFMaterialExporter.ts

```diff
@@ -1112,34 +1124,46 @@ module BABYLON.GLTF2 {
          * @param imageData map of image file name and data
          * @return glTF texture info, or null if the texture format is not supported
          */
-        private static _ExportTextureAsync(babylonTexture: BaseTexture, mimeType: ImageMimeType, images: IImage[], textures: ITexture[], samplers: ISampler[], imageData: { [fileName: string]: { data: Uint8Array, mimeType: ImageMimeType } }, useAlpha: boolean): Promise<Nullable<ITextureInfo>> {
-            const sampler = _GLTFMaterial._GetGLTFTextureSampler(babylonTexture);
-            let samplerIndex: Nullable<number> = null;
+        private _exportTextureAsync(babylonTexture: BaseTexture, mimeType: ImageMimeType): Promise<Nullable<ITextureInfo>> {
+            return new Promise((resolve, reject) => {

```

Comment:
this shouldn't be a new promise, use Promise.resolve().then() instead or the errors from _createBase64FromCanvasAsync is not captured.

Reply:
Gotcha, changed.